driver-async/src/main/com/mongodb/async/client/ClientSessionImpl.java

```diff
@@ -31,9 +31,15 @@
 
 class ClientSessionImpl extends BaseClientSessionImpl implements ClientSession {
 
+    private enum TransactionState {
+        NONE, IN, DONE, ABORTED
+    }
+
     private final OperationExecutor executor;
-    private boolean inTransaction;
+    private TransactionState transactionState = TransactionState.NONE;
     private boolean messageSent;
+    private boolean commitInProgress;
```

Comment:
I could have added this to TransactionState enum, but it didn't quite fit semantically and is not part of the state diagram.  Regardless, it's necessary to implement commitTransaction retriability.

Reply:
I can see the need for a enum state that indicates the transaction is InProgress:
```
 return transactionState == TransactionState.IN || transactionState == TransactionState.INPROGRESS;
```
Seems cleaner than:

```
return transactionState == TransactionState.IN || (transactionState == TransactionState.DONE && commitInProgress);
```

Reply:
I agree it doesn't fit well into the state diagram. I think the reason is that it is a temporary state that only lasts for the duration of the commitTransaction method.